### ISSUE TITLE
Migration to v2 causes emails to be sent when the password is migrated

### ISSUE THREAD

**Issue by IsmailM** [erdostom/devise-argon2#18]
As per https://github.com/erdostom/devise-argon2?tab=readme-ov-file#upgrade-steps, the password is silently updated in the background as part of the migration:

https://github.com/erdostom/devise-argon2/blob/db45d0dc31dcedf93189719d6877d6c36a2b4d8b/lib/devise-argon2/model.rb#L52-L58

However, if the following is set to true, then an email is sent to the end user telling that the password was updated during this silent migration.

```ruby
  # Send a notification email when the user's password is changed.
  config.send_password_change_notification = true
```

This can be pretty confusing and jarring, given that the password has only been 'updated' in the background.

Maybe it makes sense to override the method below to ignore sending the email if it is only a silent migration:

https://github.com/heartcombo/devise/blob/72884642f5700439cc96ac560ee19a44af5a2d45/lib/devise/models/database_authenticatable.rb#L167-L170

Any thoughts / suggestions?

**Comment by moritzhoeppner** [comment]
Hi @IsmailM,

Thank you for the report! My first reaction is that we should not trigger callbacks at all in `update_hash`. If that is correct, we should use `update_columns` instead of `save` for updating persisted records: https://github.com/moritzhoeppner/devise-argon2/commit/4c5ee3413051ff1de6e2376734114f92199145c5

Until we fix this issue, a quick solution for you could be to override the `update_hash` method in your user model.

**Comment by erdostom** [comment]
Thank you for raising this issue @IsmailM and @moritzhoeppner for the fix. I just released v2.0.2 that incorporates it.
